streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -208,31 +205,57 @@ public synchronized void start() {
 
     /**
      * Shutdown this stream instance by signaling all the threads to stop,
-     * and then wait for them to join.
+     * and then wait for them to join. Uses a default timeout of 30 seconds
+     */
+    public void close() {
+        close(DEFAULT_CLOSE_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Shutdown this stream instance by signaling all the threads to stop,
+     * and then wait up to the timeout for the threads to join.
```

Comment:
State that `A timeout of 0 means to wait forever`. 

1. Also, I feel it's better to use 0 as the default value in the above function to be aligned with Thread.join() semantics? 

2. Could we return a boolean indicating if the instance has been successfully closed or it is timed out, besides the warning log? Users can then program around it instead of only doing operational actions based on logs.